stormlog/derived_fields.py

```
@@ -0,0 +1,246 @@
+
+    Clamped to zero — negative values indicate a stale or inconsistent snapshot
+    rather than a meaningful metric.
+    """
+    return max(0, allocator_reserved_bytes - allocator_allocated_bytes)
+
+
+def _compute_utilization_ratio(
+    allocator_allocated_bytes: int,
+    device_total_bytes: None | int,
+) -> None | float:
+    """Fraction of device capacity currently allocated (0.0 – 1.0)."""
```

Ruff (0.15.12)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)